designer/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessStatusDefinitionService.scala

```diff
@@ -0,0 +1,43 @@
+package pl.touk.nussknacker.ui.process
+
+import io.circe.generic.JsonCodec
+import pl.touk.nussknacker.engine.ProcessingTypeData
+import pl.touk.nussknacker.engine.api.deployment.StateStatus.StateId
+import pl.touk.nussknacker.ui.process.processingtypedata.ProcessingTypeDataProvider
+
+import java.net.URI
+
+class ProcessStatusDefinitionService(typeToConfig: ProcessingTypeDataProvider[ProcessingTypeData],
+                                     categoryService: ProcessCategoryService) {
+
+  def fetchStateDefinitions(): List[StateDefinition] = {
+    //gather state definitions for all processing types
+    val allProcessingTypeStateDefinitions = typeToConfig.all.toList.flatMap { case (processingType, processingTypeData) =>
+      val processingTypeCategories = categoryService.getProcessingTypeCategories(processingType).toSet
+      val stateDefinitionManager = processingTypeData.deploymentManager.processStateDefinitionManager
+      stateDefinitionManager.statusIds().map(name =>
```

Comment:
IMO would be nice to have one public method: `stateDefinitions()` inside  `ProcessStateDefinitionManager` and other methods like `statusDisplayableName` and `statusIcon` would be only a helper methods. The only problem I can see is that you would have a tuple (statedefinition, category) but maybe clean API, with less methods is more important?

Reply:
Good point.
Added `stateDefinitions` to provide default set of states and their default properties. This "reversed" current behaviour of `ProcessStateDefinitionManager`: 

1. stateDefinitions - method returns default "fixed" set of states, used to display filtering options and used as defaults for `ProcessState` properties (which are displayed with status icon)
2. statusTooltip, statusIcon and statusDescription - by default those methods return values from "stateDefinitions", but you can override them to customize "dynamic" properties of state (which happens in e.g. `PeriodicProcessStateDefinitionManager` and other managers, that mechanism of "embelishments" does not change)

Reply:
Uhh I see now that we hadn't understand correctly each other. By "one public method" I mean one new method instead of three new methods. IMO we shouldn't generalize this status definition so much. We need to have only name, displayableName and icon in this definition, putting tooltip and description there will be clumsy for "dynamic" statuses. What description you want to provide to `Scheduled(on Friday)` state with status.name =  "SCHEDULED"?

Reply:
After discussion: 1) we do not add another generalization but reusing current one 2) this approach eliminates several mistakes in state definitions 3) part of `StateDefinitionDetails` properties are used to build filtering options (displayableName, icon, tooltip) and some are used as default ProcessState properties (icon, tooltip, description) - hence misunderstanding.